autokeras/auto/auto_model.py

```diff
@@ -4,96 +4,45 @@
 import tensorflow as tf
 from tensorflow.python.util import nest
 
+from autokeras.hypermodel import hyper_block
 from autokeras.hypermodel import hyper_head
 from autokeras import layer_utils
 from autokeras import const
 
 
-class AutoModel(kerastuner.HyperModel):
-    """ A AutoModel should be an AutoML solution.
+class GraphAutoModel(kerastuner.HyperModel):
+    """A HyperModel defined by a graph of HyperBlocks.
 
-    It contains the HyperModels and the Tuner.
+    GraphAutoModel is a subclass of HyperModel. Besides the HyperModel properties,
+    it also has a tuner to tune the HyperModel. The user can use it in a similar
+    way to a Keras model since it also has `fit()` and  `predict()` methods.
+
+    The user can specify the high-level neural architecture by connecting the
+    HyperBlocks with the functional API, which is the same as
+    the Keras functional API.
 
     Attributes:
-        inputs: A HyperModel instance. The input node of a the AutoModel.
-        outputs: A HyperModel instance. The output node of the AutoModel.
-        hypermodel: An instance of HyperModelWrap connecting from the inputs to the
-            outputs.
-        tuner: An instance of Tuner.
+        inputs: A list of or a HyperNode instances.
+            The input node(s) of the GraphAutoModel.
+        outputs: A list of or a HyperNode instances.
+            The output node(s) of the GraphAutoModel.
+        trials: Int. The number of different models to try.
```

Comment:
Should this be renamed to `max_trials` across the board? It may not be an exact number, but rather an upper bound, because the tuner or oracle may interrupt the tuning process (e.g. if the search space is too small).